plugins/ingest/src/test/java/org/elasticsearch/ingest/TestTemplateService.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest;
+
+import java.util.Map;
+
+public class TestTemplateService implements TemplateService {
+
+    private final static TemplateService INSTANCE = new TestTemplateService();
+
+    public static TemplateService instance() {
+        return INSTANCE;
+    }
+
+    public static Template template(String expected) {
```

Comment:
I did this to avoid using class names, but `new TestTemplateService.MockTemplate(...)` isn't that longer than `TestTemplateService.template(...)`. I'll remove it.